deploy/metrics/loki_exporter.py

```
@@ -0,0 +1,278 @@
+            self.flush()
+
+    @staticmethod
+    def _to_nanoseconds(timestamp: Optional[str]) -> int:
+        """Convert an ISO timestamp to nanoseconds since epoch, or use current time."""
+        if not timestamp:
+            return int(time.time() * 1e9)
+        try:
+            from datetime import datetime, timezone
+            dt = datetime.fromisoformat(timestamp.replace("Z", "+00:00"))
+            return int(dt.timestamp() * 1e9)
+        except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)